bundles/org.openhab.binding.dbquery/src/main/java/org/openhab/binding/dbquery/internal/Value2StateConverter.java

```diff
@@ -0,0 +1,139 @@
+                if (bool != null)
+                    return bool ? OpenClosedType.OPEN : OpenClosedType.CLOSED;
+                else
+                    return UnDefType.NULL;
+            } else {
+                throw new UnnexpectedCondition("Not expected targetType " + targetType);
+            }
+        }
+    }
+
+    private State convert2DateTime(Object value) {
+        if (value instanceof Instant)
+            return new DateTimeType(ZonedDateTime.ofInstant((Instant) value, ZoneId.systemDefault()));
+        else if (value instanceof Date)
+            return new DateTimeType(ZonedDateTime.ofInstant(((Date) value).toInstant(), ZoneId.systemDefault()));
+        else if (value instanceof String)
+            return new DateTimeType((String) value);
+        else {
+            logger.warn("Can't convert {} to DateTimeType", value);
+            return UnDefType.NULL;
+        }
+    }
+
+    private State convert2Decimal(Object value) {
+        if (value instanceof Integer)
+            return new DecimalType((Integer) value);
+        else if (value instanceof Long)
+            return new DecimalType((Long) value);
+        else if (value instanceof Float)
+            return new DecimalType((Float) value);
+        else if (value instanceof Double)
+            return new DecimalType((Double) value);
+        else if (value instanceof BigDecimal)
+            return new DecimalType((BigDecimal) value);
+        else if (value instanceof BigInteger)
+            return new DecimalType(new BigDecimal((BigInteger) value));
+        else if (value instanceof Number)
+            return new DecimalType(((Number) value).longValue());
+        else if (value instanceof String)
+            return DecimalType.valueOf((String) value);
+        else if (value instanceof Duration)
+            return new DecimalType(((Duration) value).toMillis());
+        else {
+            logger.warn("Can't convert {} to DecimalType", value);
+            return UnDefType.NULL;
+        }
+    }
+
+    private State convert2String(Object value) {
+        if (value instanceof String) {
+            return new StringType((String) value);
+        } else if (value instanceof byte[]) {
+            return new StringType(Base64.getEncoder().encodeToString((byte[]) value));
+        } else if (value instanceof QueryResult) {
+            return new StringType(jsonEncoder.encode((QueryResult) value));
+        } else {
+            return new StringType(String.valueOf(value));
+        }
+    }
+
+    private @Nullable Boolean convert2Boolean(Object value) {
+        if (value instanceof Boolean)
+            return (Boolean) value;
+        else if (value instanceof Number)
+            return ((Number) value).doubleValue() != 0d;
+        else if (value instanceof String) {
+            var svalue = (String) value;
+            return Boolean.parseBoolean(svalue) || (svalue.equalsIgnoreCase("on")) || svalue.equals("1");
```

Comment:
The code analysis reports this as a WARNING.

svalue.equals("1")

should be "1".equals(svalue)
reason is only the value in the brackets is checked safely for NULL and wont throw a NPE. Please change all occurrences of this to remove the warnings.

Reply:
ignore then as I have not compiled it, only guessed this would be an issue for the code analysis report that is found in the target folder. If it does not show up in the report then ignore.